bionic/aip/task.py

```diff
@@ -0,0 +1,156 @@
+    worker_machine: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class Job:
+    """A job groups together individual tasks, all of which share these properties
+
+    Attributes
+    ----------
+    uuid: str
+        The globally unique job name on AI platform
+    image: str
+        The name of the docker image for the tasks
+    project: str
+        The GCP project where the jobs will be run
+
+    """
+
+    uuid: str
+    image: str
+    project: str
+    account: str = None
+    network: str = None
+
+    def __post_init__(self):
+        if "gcr.io" not in self.image:
+            # NOTE this is just for convenient image shorthand
+            object.__setattr__(self, "image", f"gcr.io/{self.project}/{self.image}")
+
+
+@dataclass(frozen=True)
+class Task:
+    name: str
+    job: Job
+    function: Callable
+    resource: Resource
+    tune: Optional[Tune] = None
+
+    def job_id(self):
+        return f"{self.job.uuid}_{self.name}"
+
+    def path_inputs(self):
+        # TODO bionic will need to customize
+        return f"gs://{self.job.project}/bionic/{self.job.uuid}/{self.name}-inputs.pkl"
+
+    def path_output(self):
+        # TODO bionic will need to customize
+        return f"gs://{self.job.project}/bionic/{self.job.uuid}/{self.name}-output.pkl"
+
+    def _stage(self):
+        path = self.path_inputs()
+        logging.info(f"Staging task {self.name} at {path}")
+        with blocks.filesystem.GCSNativeFileSystem().open(path, "wb") as f:
+            cloudpickle.dump(self, f)
+
+    def _ai_platform_job_spec(self):
+        """Conversion from our task data model to a job request on ai platform"""
+        output = {
+            "jobId": f"{self.job.uuid}_{self.name}",
+            "labels": {"job": self.job.uuid},
+            "trainingInput": {
+                "serviceAccount": self.job.account,
+                "masterType": self.resource.machine,
+                "masterConfig": {"imageUri": self.job.image},
+                "args": ["python", "-m", "bionic.aip.task", self.path_inputs()],
+                "packageUris": [],
+                "region": "us-west1",
+                "pythonModule": "",
+                "scaleTier": "CUSTOM",
+            },
+        }
+        if self.job.network is not None:
+            output["trainingInput"]["network"] = self.job.network
+
+        if self.resource.worker_count is not None and self.resource.worker_count > 0:
+            output["trainingInput"]["workerCount"] = self.resource.worker_count
+            output["trainingInput"]["workerType"] = self.resource.worker_machine
+            output["trainingInput"]["workerConfig"] = {"imageUri": self.job.image}
+
+        if self.tune is not None:
+            output["trainingInput"]["hyperparameters"] = self.tune.spec()
+
+        return output
+
+    def submit(self) -> Future:
+        self._stage()
+        spec = self._ai_platform_job_spec()
+
+        if os.environ.get("TEST", "").lower() == "true":
```

Comment:
We use pytest for all our Bionic testing. Can you replace this environment variable with pytest variable `PYTEST_CURRENT_TEST`? You can find more information here https://docs.pytest.org/en/latest/example/simple.html#pytest-current-test-environment-variable

Reply:
yeah i had actually intended to remove this whole code path, this was just for testing the demo